src/compass/s1_geocode_slc.py

```diff
@@ -139,28 +124,56 @@ def run(cfg: GeoRunConfig):
-                                      iters, blocksize, flatten,
-                                      azimuth_carrier=az_carrier_poly2d,
-                                      az_time_correction=az_lut,
-                                      srange_correction=rg_lut)
+            check_eap = is_eap_correction_necessary(burst.ipf_version)
+            for b in bursts:
+                pol = b.polarization
+
+                # Load the input burst SLC
+                temp_slc_path = f'{scratch_path}/{out_paths.file_name_stem}_{pol}_temp.vrt'
+                burst.slc_to_vrt_file(temp_slc_path)
+
+                # Apply EAP correction if necessary
+                if check_eap.phase_correction:
+                    temp_slc_path_corrected = temp_slc_path.replace('_temp.vrt',
+                                                                    '_corrected_temp.rdr')
+                    apply_eap_correction(b,
+                                         temp_slc_path,
+                                         temp_slc_path_corrected,
+                                         check_eap)
+
+                    # Replace the input burst if the correction is applied
+                    temp_slc_path = temp_slc_path_corrected
+
+
+                # Split the range bandwidth of the burst, if required
+                if cfg.split_spectrum_params.enabled:
+                    rdr_burst_raster = range_split_spectrum(b,
+                                                            temp_slc_path,
+                                                            cfg.split_spectrum_params,
+                                                            scratch_path)
+                else:
+                    rdr_burst_raster = isce3.io.Raster(temp_slc_path)
+
+                init_geocoded_dataset(grid_group, pol, geo_grid, 'complex64',
+                                      f'{pol} geocoded CSLC image')
+
+                # access the HDF5 dataset for a given frequency and polarization
+                dataset_path = f'{grid_path}/{pol}'
+                gslc_dataset = geo_burst_h5[dataset_path]
+
+                # Construct the output raster directly from HDF5 dataset
+                geo_burst_raster = isce3.io.Raster(f"IH5:::ID={gslc_dataset.id.id}".encode("utf-8"),
+                                                   update=True)
+
+                # Geocode
+                isce3.geocode.geocode_slc(geo_burst_raster, rdr_burst_raster,
+                                          dem_raster,
+                                          radar_grid, sliced_radar_grid,
+                                          geo_grid, orbit,
+                                          native_doppler,
+                                          image_grid_doppler, ellipsoid, threshold,
+                                          iters, blocksize, flatten,
+                                          azimuth_carrier=az_carrier_poly2d)
```

Comment:
Can you please update this branch and have it in sync with the main repository? I do not see the LUT allocated here